src/app/client/src/app/plugins/profile/components/submit-teacher-details/submit-teacher-details.component.ts

```diff
@@ -167,12 +180,110 @@ export class SubmitTeacherDetailsComponent implements OnInit, OnDestroy {
     this.enableSubmitBtn = (this.userDetailsForm.status === 'VALID');
+    this.setFormData();
     this.getState();
     this.showLoader = false;
     this.onStateChange();
     this.enableSubmitButton();
   }
 
+  setFormData() {
+    // TODO: handle fetching of data from API for update form
+    const email = this.userProfile.maskedEmail;
+    const phone = this.userProfile.maskedPhone;
+    if (email) {
+      this.userDetailsForm.controls['email'].setValue(email);
+    }
+    if (phone) {
+      this.userDetailsForm.controls['phone'].setValue(phone);
+    }
+    const fieldType = ['email', 'phone'];
+    for (let index = 0; index < fieldType.length; index++) {
+      const key = fieldType[index];
+      const keyControl = this.userDetailsForm.controls[key];
+      const email = this.userProfile['masked' + key];
+      keyControl.valueChanges.pipe(debounceTime(400), distinctUntilChanged()).subscribe((newValue) => {
+        if (email === newValue) {
+          this.validationType[key].isVerified = false;
+          this.validationType[key].isVerificationRequired = false;
+          this.userDetailsForm.removeControl(key + 'Verified');
+          return;
+        }
+        if (newValue && keyControl.status === 'VALID') {
+          this.userDetailsForm.addControl(key + 'Verified', new FormControl('', Validators.required));
+          this.userDetailsForm.controls[key + 'Verified'].setValue('');
+          this.validationType[key].isVerified = false;
+          this.validationType[key].isVerificationRequired = true;
+        } else if (!newValue) {
+          this.validationType[key].isVerified = false;
+          this.validationType[key].isVerificationRequired = false;
+          this.userDetailsForm.removeControl(key + 'Verified');
+        }
+      });
+    }
+  }
+
+  validateUser(fieldType) {
+    this.generateOTP(fieldType);
+  }
+
+  generateOTP(fieldType) {
+    const request = {
+      request: {
+        key: fieldType === 'phone' ?
+          this.userDetailsForm.controls.phone.value.toString() : this.userDetailsForm.controls.email.value,
+        type: fieldType
+      }
+    };
+    this.otpService.generateOTP(request).subscribe((data: ServerResponse) => {
+        this.otpData = this.prepareOtpData(fieldType);
+        this.isOtpVerificationRequired = true;
+      },
+      (err) => {
+        this.toasterService.error(this.resourceService.messages.fmsg.m0051);
+      }
+    );
+  }
+
+  onVerificationSuccess(data) {
+    this.isOtpVerificationRequired = false;
+    const fieldType = this.getFieldType(data);
+    this.validationType[fieldType].isVerified = true;
+    this.userDetailsForm.controls[fieldType + 'Verified'].setValue(true);
+    this.validationType[fieldType].isVerificationRequired = false;
+  }
+
+  getFieldType(data) {
+    return _.get(data, 'phone') ? 'phone' : 'email';
+  }
+
+  onOtpPopupClose() {
+    this.isOtpVerificationRequired = false;
```

Comment:
make it seperate function 
function  (value = false)   this.isOtpVerificationRequired = value;